### DISCUSSION TITLE
NXP HAL S32: Possibility of changing SOC clock configuration

### DISCUSSION THREAD

**Discussion by vChavezB** [zephyrproject-rtos/zephyr#74153] [Ideas] [answered]
By default the S32K146 SOC has a fixed clock configuration which is selected [here](https://github.com/zephyrproject-rtos/hal_nxp/blob/bee15c31e19ff1982583d9b750ef844d64320160/s32/CMakeLists.txt#L15) which then compiles the clock configuration (here is the S32K146 for [example](https://github.com/zephyrproject-rtos/hal_nxp/blob/bee15c31e19ff1982583d9b750ef844d64320160/s32/soc/s32k146/src/Clock_Ip_Cfg.c))

It would be nice if the the variable `Clock_Ip_aClockConfig` would be declared `weak`.

This allows to override the clock configuration and not change the source code for hal_nxp upstream for each different clock configuration I need.

Use case: I have a S32 custom board that I need to change clock speed, SPLL source ,etc. for my project requirements.

**Comment by manuargue** [comment, marked as answer]
Thanks. It makes sense as a workaround for now. Ideally the clock tree should be described in devicetree to allow reconfiguration, but there are limitations in the HAL API design that prevents this at some extent. I will give it a try and submit a patch for this if all good.

**Reply by vChavezB** [reply]
Thanks for checking out. I see your point about the device tree, that would be more flexible 👍
